hermes-common/src/main/java/pl/allegro/tech/hermes/infrastructure/zookeeper/ZookeeperBasedRepository.java

```diff
@@ -156,6 +166,20 @@ protected void createInTransaction(String path, Object value, String childPath)
                 .commit();
     }
 
+    protected void deleteInTransaction(List<String> paths) throws Exception {
+        if (paths.isEmpty()) {
+            throw new InternalProcessingException("Attempting to remove empty set of paths from ZK");
+        }
+        ensureConnected();
+        CuratorTransactionFinal transaction = zookeeper.inTransaction().delete().forPath(paths.get(0)).and();
+
+        for (int i = 1; i < paths.size(); i++) {
+            transaction = transaction.delete().forPath(paths.get(i)).and();
+        }
+
+        transaction.commit();
+    }
```

Comment:
**note:** Here is my proposition for refactor:

```java
protected void deleteInTransaction(List<String> paths) throws Exception {
    if (paths.isEmpty()) {
        throw new InternalProcessingException("Attempting to remove empty set of paths from ZK");
    }
    ensureConnected();
    zookeeper.transaction().forOperations(
            paths.stream().map(x -> {
                try {
                    return zookeeper.transactionOp().delete().forPath(x);
                } catch (Exception e) {
                    throw new RuntimeException(e);
                }
            }).collect(Collectors.toList())
    );
}
```

Feel free to tweak with it :)

Reply:
Will be resolved here: https://github.com/allegro/hermes/issues/1886